addons/account/models/res_partner_bank.py

```diff
@@ -63,6 +64,21 @@ def _check_allow_out_payment(self):
                 if not self.env.user.has_group('account.group_validate_bank_account'):
                     raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
 
+    @api.depends('acc_number')
+    def _compute_alert_duplicate_bank_account(self):
+        """Add a warning if the bank account is already in use elsewhere"""
+        acc_numbers = [bank.acc_number for bank in self if bank.acc_number]
+        duplicates = self.env['res.partner.bank'].search([('acc_number', 'in', acc_numbers)])
```

Comment:
```suggestion
        duplicates = self.env['res.partner.bank'].search([('acc_number', 'in', self.mapped('acc_number'))]).grouped('acc_number')
```
This is easier for the next part.